socorro/cron/jobs/ftpscraper.py

```diff
@@ -71,25 +80,46 @@ def patient_urlopen(url, max_attempts=4, sleep_time=20):
 
 class ScrapersMixin(object):
 
-    def getLinks(self, url, startswith=None, endswith=None):
+    def get_links(self, url, startswith=None, endswith=None):
 
         html = ''
         results = []
         content = patient_urlopen(url, sleep_time=30)
         if not content:
             return []
+
         html = lxml.html.document_fromstring(content)
 
-        for element, attribute, link, pos in html.iterlinks():
-            if startswith:
+        def url_match(link):
+            # The link might be something like "/pub/mobile/nightly/"
+            # but we're looking for a path that starts with "nightly".
+            # So first we need to remove what's part of the base URL
+            # to make a fair comparison.
+            if startswith is not None:
+                # If the current URL is http://example.com/some/dir/
+                # and the link is /some/dir/mypage/ and the thing
+                # we're looking for is "myp" then this should be true
+                path = urlparse.urlparse(url).path
+                if link.startswith(path):
+                    link = link.replace(path, '')
                 if link.startswith(startswith):
-                    results.append(link)
+                    return True
             elif endswith:
                 if link.endswith(endswith):
-                    results.append(link)
+                    return True
+            else:
+                raise NotImplementedError(
```

Comment:
on doing that suggestion, the code could further simplified from line 105 onward to just:
``` python
                return link.startswith(startswith)
            return link.endswith(endswith)
```
Naming things is something that we often disagree on, but having the variable names match the function names is unnecessary visual overloading.  Please consider changing the variables to something like ```prefix``` & ```suffix``` or ```start_string``` & ```end_string```